src/options.js

```diff
@@ -575,3 +576,35 @@ export function maybeClip(clip) {
   else if (clip != null) clip = keyword(clip, "clip", ["frame", "sphere"]);
   return clip;
 }
+
+// Duck typing Apache Arrow tables
+function isArrowTable(data) {
+  return typeof data?.getChild === "function" && typeof data.numRows === "number" && typeof data.slice === "function";
+}
+
+// Extract columnar data
+function columnar(data, name, type) {
+  if (isArrowTable(data)) {
+    const column = maybeTypedArrayify(data.getChild(name), type);
```

Comment:
I think this should be calling `data.getChild(name).toArray()` here, or else `maybeTypedArrayify` will use the slowest iterable path rather than using the more efficient `Vector.toArray` (which could be zero-copy if the Arrow table only has one chunk).